src/main/java/org/parchmentmc/lodestone/util/OfflineChecker.java

```
@@ -0,0 +1,13 @@
+package org.parchmentmc.lodestone.util;
+
+import org.gradle.api.Project;
+
+public class OfflineChecker {
+    public static void checkOffline(Project project)
+    {
+        if (project.getGradle().getStartParameter().isOffline())
+        {
+            throw new IllegalStateException("Gradle is offline. Cannot download minecraft metadata.");
+        }
```

I want to note, in my opinion Lodestone should still be able to function even if running offline, if the manifest files have already been downloaded previously. Lodestone should only error out when offline mode is enabled if the manifest files are missing, or possibly massively out-of-date (like, months old).

Your choice if you want to try your hand at this or not, this is just a gripe I have about this.

---

That's exactly why I moved it into the task action instead of the constructor. It means if the task is UP-TO-DATE then this code will never be called.